diff --git a/Code/ForceField/MMFF/StretchBend.h b/Code/ForceField/MMFF/StretchBend.h
index 4eceb3e7e..765fabde6 100644
--- a/Code/ForceField/MMFF/StretchBend.h
+++ b/Code/ForceField/MMFF/StretchBend.h
@@ -10,6 +10,7 @@
 #ifndef __RD_MMFFSTRETCHBEND_H__
 #define __RD_MMFFSTRETCHBEND_H__
 
+#include <cstdint>
 #include <utility>
 #include <vector>
 
diff --git a/Code/ForceField/MMFF/TorsionAngle.h b/Code/ForceField/MMFF/TorsionAngle.h
index 2fbab767f..0ffe69833 100644
--- a/Code/ForceField/MMFF/TorsionAngle.h
+++ b/Code/ForceField/MMFF/TorsionAngle.h
@@ -12,6 +12,7 @@
 #define RD_MMFFTORSIONANGLE_H
 
 #include <ForceField/Contrib.h>
+#include <cstdint>
 #include <tuple>
 #include <vector>
 
