diff --git a/Code/GraphMol/Chirality.cpp b/Code/GraphMol/Chirality.cpp
index cf500a932..18272eb0c 100644
--- a/Code/GraphMol/Chirality.cpp
+++ b/Code/GraphMol/Chirality.cpp
@@ -434,6 +434,7 @@ std::optional<Atom::ChiralType> atomChiralTypeFromBondDirPseudo3D(
                "bad bond direction");
   constexpr double coordZeroTol = 1e-4;
   constexpr double zeroTol = 1e-3;
+  constexpr double tShapeTol = 0.01;  // used to recognize T-shaped arrangements
 
   // NOTE that according to the CT file spec, wedging assigns chirality
   // to the atom at the point of the wedge, (atom 1 in the bond).
@@ -756,7 +757,7 @@ std::optional<Atom::ChiralType> atomChiralTypeFromBondDirPseudo3D(
     auto crossp1 = bv1.crossProduct(bv2);
     // catch linear arrangements
     if (nNbrs == 3) {
-      if (crossp1.lengthSq() < 5 * zeroTol) {
+      if (crossp1.lengthSq() < tShapeTol) {
         // in a linear relationship with three neighbors we assume that the
         // two perpendicular bonds are wedged in the other direction from the
         // one that was provided.
