diff --git a/External/ChemDraw/CMakeLists.txt b/External/ChemDraw/CMakeLists.txt
index b75d32795..f745c94c6 100644
--- a/External/ChemDraw/CMakeLists.txt
+++ b/External/ChemDraw/CMakeLists.txt
@@ -55,6 +55,11 @@ if(RDK_BUILD_CHEMDRAW_SUPPORT)
   # however, we don't want to install it so use the undocumented EXCLUDE_FROM_ALL
   add_subdirectory(chemdraw/expatpp EXCLUDE_FROM_ALL)
 
+  # our windows build has trouble linking to expatpp because of the expatpp.h
+  # include. Make the path to it a public dependency so that things that
+  # depend on it can find it.
+  target_include_directories(expatpp PUBLIC ${EXPAT_INCLUDE_DIR})
+
   include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}/chemdraw/expatpp/expatpp-code-r6-trunk/src_pp
     ${CMAKE_CURRENT_SOURCE_DIR}/chemdraw/expatpp/expatpp-code-r6-trunk/expat/lib)
@@ -65,6 +70,7 @@ if(RDK_BUILD_CHEMDRAW_SUPPORT)
   target_compile_definitions(ChemDraw PRIVATE CHEMDRAW_BUILD)
   target_link_libraries(ChemDraw PRIVATE expatpp)
 
+
   # export all the symbols for ChemDraw on MSVC
   if((MSVC AND RDK_INSTALL_DLLS_MSVC) OR((NOT MSVC) AND WIN32))
     message("== ChemDraw exporting all symbols")
