diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1abad9260..0994f939a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,7 +72,7 @@ option(RDK_BUILD_MINIMAL_LIB_RXN "build support for reactions into MinimalLib" O
 option(RDK_BUILD_MINIMAL_LIB_SUBSTRUCTLIBRARY "build support for SubstructLibrary into MinimalLib" ON )
 option(RDK_BUILD_MINIMAL_LIB_MCS "build support for MCS into MinimalLib" OFF )
 
-set(RDK_BOOST_VERSION "1.70.0")
+set(RDK_BOOST_VERSION "1.81.0")
 
 if(NOT MSVC)
   if(RDK_OPTIMIZE_POPCNT)
@@ -383,7 +383,7 @@ if(RDK_BUILD_PYTHON_WRAPPERS)
   endif()
 
 else(RDK_BUILD_PYTHON_WRAPPERS)
-  find_package(Boost ${RDK_BOOST_VERSION} REQUIRED CONFIG)
+  find_package(Boost ${RDK_BOOST_VERSION} REQUIRED CONFIG NO_CMAKE_FIND_ROOT_PATH)
 endif(RDK_BUILD_PYTHON_WRAPPERS)
 
 find_package(Eigen3)
