diff --git a/Code/GraphMol/GaussianShape/ShapeInput.cpp b/Code/GraphMol/GaussianShape/ShapeInput.cpp
index d25d5b434..e482efe85 100644
--- a/Code/GraphMol/GaussianShape/ShapeInput.cpp
+++ b/Code/GraphMol/GaussianShape/ShapeInput.cpp
@@ -1062,6 +1062,10 @@ double maxScore(const double v1, const double v2, const double c1,
   // is v1, and the opposite.
   auto maxPart = [](const double p1, const double p2,
                     const ShapeOverlayOptions &ovlyOpts) -> double {
+    // If either is 0, the maximum overlap must be 0.
+    if (p1 == 0.0 || p2 == 0.0) {
+      return 0.0;
+    }
     if (p1 < p2) {
       return p1 / (ovlyOpts.simBeta * (p2 - p1) + p1);
     }
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSet.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSet.cpp
index c7da2adfe..ac18968a2 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSet.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSet.cpp
@@ -383,9 +383,10 @@ void SynthonSet::makeSynthonSearchMols() {
       std::vector<std::unique_ptr<ROMol>> molFrags;
       MolOps::getMolFrags(*fragMol, molFrags, false);
       int fragWeWant = findMolNumFrag(molFrags, synthSetNum);
-      unsigned int otf;
-      sanitizeMol(*static_cast<RWMol *>(molFrags[fragWeWant].get()), otf,
-                  MolOps::SANITIZE_SYMMRINGS);
+      if (!molFrags[fragWeWant]->getRingInfo()->isInitialized()) {
+        VECT_INT_VECT arings;
+        MolOps::symmetrizeSSSR(*molFrags[fragWeWant], arings);
+      }
       d_synthons[synthSetNum][j].second->setSearchMol(
           std::move(molFrags[fragWeWant]));
     }
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.cpp
index 6ba3d3969..f8366e945 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpace.cpp
@@ -322,7 +322,6 @@ SearchResults SynthonSpace::shapeSearch(
   } else {
     queryCp.reset(new RWMol(query));
   }
-
   SynthonSpaceShapeSearcher ssss(*queryCp, params, this);
   return ssss.search(ThreadMode::ThreadFragments);
 }
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceRascalSearcher.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceRascalSearcher.cpp
index f59fab72f..ef23a2c6d 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceRascalSearcher.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceRascalSearcher.cpp
@@ -135,9 +135,10 @@ bool SynthonSpaceRascalSearcher::extraSearchSetup(
   int numDone = 100;
   for (const auto &fragSet : fragSets) {
     for (const auto &frag : fragSet) {
-      unsigned int otf;
-      sanitizeMol(*static_cast<RWMol *>(frag.get()), otf,
-                  MolOps::SANITIZE_SYMMRINGS);
+      if (!frag->getRingInfo()->isInitialized()) {
+        VECT_INT_VECT arings;
+        MolOps::symmetrizeSSSR(*frag, arings);
+      }
       --numDone;
       if (!numDone) {
         numDone = 100;
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.cpp
index e0e257812..711664698 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.cpp
@@ -934,9 +934,10 @@ std::map<std::string, std::vector<ROMol *>> mapFragsBySmiles(
         return fragSmiToFrag;
       }
       // For the fingerprints and shapes, ring info is required.
-      unsigned int otf;
-      sanitizeMol(*static_cast<RWMol *>(frag.get()), otf,
-                  MolOps::SANITIZE_SYMMRINGS);
+      if (!frag->getRingInfo()->isInitialized()) {
+        VECT_INT_VECT arings;
+        MolOps::findSSSR(*frag, arings);
+      }
       std::string fragSmi = MolToSmiles(*frag);
       if (auto it = fragSmiToFrag.find(fragSmi); it == fragSmiToFrag.end()) {
         fragSmiToFrag.emplace(fragSmi, std::vector<ROMol *>(1, frag.get()));
@@ -1085,7 +1086,7 @@ std::vector<std::unique_ptr<RWMol>> generateIsomerConformers(
     EnumerateStereoisomers::StereoisomerEnumerator enu(mol, enumOpts);
     unsigned int i = 0;
     while (auto isomer = enu.next()) {
-      confMols.emplace_back(static_cast<RWMol *>(isomer.release()));
+      confMols.emplace_back(reinterpret_cast<RWMol *>(isomer.release()));
       if (++i == maxStereoCenters) {
         break;
       }
@@ -1158,7 +1159,7 @@ void splitDummyDummyBonds(RWMol &mol) {
 }
 }  // namespace
 
-std::unique_ptr<RWMol> trimSampleMol(const ROMol &mol, size_t molNum) {
+std::unique_ptr<RWMol> trimSampleMol(ROMol &mol, size_t molNum) {
   auto ts = MolToCXSmiles(mol);
   boost::dynamic_bitset<> molNumAtoms(mol.getNumAtoms());
   unsigned int molNumProp;
@@ -1171,7 +1172,7 @@ std::unique_ptr<RWMol> trimSampleMol(const ROMol &mol, size_t molNum) {
 
   auto ringInfo = mol.getRingInfo();
   if (!ringInfo->isInitialized()) {
-    MolOps::findSSSR(mol);
+    MolOps::symmetrizeSSSR(mol);
   }
   for (const auto &atomRing : ringInfo->atomRings()) {
     for (auto a : atomRing) {
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.h b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.h
index dfb9371c7..4fecd8c39 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.h
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearch_details.h
@@ -206,7 +206,7 @@ generateIsomerConformers(
 // Trim the molecule down to the minimum needed to do a sensible conformation
 // expansion for the atoms with property molNum=molNum.
 RDKIT_SYNTHONSPACESEARCH_EXPORT std::unique_ptr<RWMol> trimSampleMol(
-    const ROMol &mol, size_t molNum);
+    ROMol &mol, size_t molNum);
 
 // Make the synthon shapes for the given molecules based on the passed in
 // synthons.  The synthon of interest in each molecule is from the
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearcher.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearcher.cpp
index 4b6deb5e0..2bd43bb84 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearcher.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSearcher.cpp
@@ -651,10 +651,6 @@ void SynthonSpaceSearcher::buildAllHits(
     return;
   }
 
-  std::uint64_t numPossHits = 0;
-  for (const auto &hitset : hitsets) {
-    numPossHits += hitset->numHits;
-  }
   if (!getParams().possibleHitsFile.empty()) {
     // The file will be appended to, so clear it out at the start.
     std::ofstream ofs(getParams().possibleHitsFile);
@@ -881,11 +877,8 @@ void SynthonSpaceSearcher::sortToTryByApproxSimilarity(
   std::vector<std::pair<const SynthonSpaceHitSet *, std::vector<size_t>>>
       newToTry;
   newToTry.reserve(tmp.size());
-  std::transform(
-      tmp.begin(), tmp.end(),
-      back_inserter(newToTry), [&](const auto &p) -> auto{
-        return toTry[p.first];
-      });
+  std::transform(tmp.begin(), tmp.end(), back_inserter(newToTry),
+                 [&](const auto &p) -> auto { return toTry[p.first]; });
   toTry = newToTry;
 }
 
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceShapeSearcher.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceShapeSearcher.cpp
index 9f90c9a70..fe5cd2388 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceShapeSearcher.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceShapeSearcher.cpp
@@ -329,9 +329,10 @@ std::map<std::string, std::vector<ROMol *>> mapFragsByAtoms(
         return atomsToFrags;
       }
       // Ring info is required.
-      unsigned int otf;
-      sanitizeMol(*static_cast<RWMol *>(frag.get()), otf,
-                  MolOps::SANITIZE_SYMMRINGS);
+      if (!frag->getRingInfo()->isInitialized()) {
+        VECT_INT_VECT arings;
+        MolOps::symmetrizeSSSR(*frag, arings);
+      }
       std::vector<unsigned int> atIdxs;
       unsigned int dummyIdx = 10000;
       for (const auto a : frag->atoms()) {
@@ -944,16 +945,19 @@ bool checkExcludedVols(const ROMol &mol, const SynthonSpaceSearchParams &params,
   if (params.excludedVolume) {
     excludedVol = calcExcludedVolume(mol, *params.excludedVolume,
                                      params.shapeOverlayOptions);
-    auto numClashes = calcNumClashes(mol, *params.excludedVolume);
-    meanExcludedVol = excludedVol / static_cast<double>(numClashes);
     if (params.maxExcludedVolume > -0.5 &&
         excludedVol > params.maxExcludedVolume) {
       return false;
     }
-    if (params.maxMeanExcludedVolume > -0.5) {
-      if (meanExcludedVol > params.maxMeanExcludedVolume) {
+    auto numClashes = calcNumClashes(mol, *params.excludedVolume);
+    if (numClashes) {
+      meanExcludedVol = excludedVol / static_cast<double>(numClashes);
+      if (params.maxMeanExcludedVolume > -0.5 &&
+          meanExcludedVol > params.maxMeanExcludedVolume) {
         return false;
       }
+    } else {
+      meanExcludedVol = 0.0;
     }
   }
   return true;
@@ -970,20 +974,19 @@ bool finaliseHit(GaussianShape::ShapeInput &hitShapes,
                  double &meanExcludedVol) {
   // Copy the conformer into the hit.
   hitShapes.setActiveShape(hitShapeNum);
-  hit = static_cast<ROMol>(*hitShapes.shapeToMol(false, true));
-  if (!checkExcludedVols(hit, params, excludedVol, meanExcludedVol)) {
+  std::unique_ptr<ROMol> possHit = hitShapes.shapeToMol(false, true);
+  if (!checkExcludedVols(*possHit, params, excludedVol, meanExcludedVol)) {
     return false;
   }
+  hit = std::move(*possHit);
   hit.setProp<double>("Similarity", scores[0]);
   hit.setProp<double>("ShapeScore", scores[1]);
   hit.setProp<double>("ColorScore", scores[2]);
   const auto prodName = details::buildProductName(rxnId, synthNames);
   hit.setProp<std::string>(common_properties::_Name, prodName);
   MolOps::assignStereochemistryFrom3D(hit);
-  if (excludedVol > -0.5) {
+  if (params.excludedVolume) {
     hit.setProp<double>("ExcludedVolume", excludedVol);
-  }
-  if (meanExcludedVol > -0.5) {
     hit.setProp<double>("MeanExcludedVolume", meanExcludedVol);
   }
   return true;
@@ -995,16 +998,15 @@ bool finaliseHit(GaussianShape::ShapeInput &hitShapes,
 void finaliseHit(const ROMol &hit, const std::array<double, 3> &scores,
                  const std::string &rxnId,
                  const std::vector<const std::string *> &synthNames,
+                 const SynthonSpaceSearchParams &params,
                  const double excludedVol, const double meanExcludedVol) {
   hit.setProp<double>("Similarity", scores[0]);
   hit.setProp<double>("ShapeScore", scores[1]);
   hit.setProp<double>("ColorScore", scores[2]);
   const auto prodName = details::buildProductName(rxnId, synthNames);
   hit.setProp<std::string>(common_properties::_Name, prodName);
-  if (excludedVol > -0.5) {
+  if (params.excludedVolume) {
     hit.setProp<double>("ExcludedVolume", excludedVol);
-  }
-  if (meanExcludedVol > -0.5) {
     hit.setProp<double>("MeanExcludedVolume", meanExcludedVol);
   }
 }
@@ -1129,6 +1131,7 @@ bool SynthonSpaceShapeSearcher::verifyHit(
     ROMol &hit, const std::string &rxnId,
     const std::vector<const std::string *> &synthNames) {
   std::array<double, 3> initScores{-1.0, -1.0, -1.0};
+  std::unique_ptr<ROMol> initHit;
   if (hit.getNumConformers()) {
     initScores = GaussianShape::AlignMolecule(
         dp_queryShape->getShapes(), hit, GaussianShape::ShapeInputOptions(),
@@ -1137,12 +1140,16 @@ bool SynthonSpaceShapeSearcher::verifyHit(
     if (checkExcludedVols(hit, getParams(), excludedVol, meanExcludedVol)) {
       // If the excluded vol is also ok, take this hit.  Otherwise, pass it
       // through to conformation expansion to see if it wins there.
-      finaliseHit(hit, initScores, rxnId, synthNames, excludedVol,
+      finaliseHit(hit, initScores, rxnId, synthNames, getParams(), excludedVol,
                   meanExcludedVol);
       if (!getParams().bestHit && checkBondLengths(hit) &&
           initScores[0] >= getParams().similarityCutoff) {
+        updateBestHitSoFar(hit, initScores[0]);
         return true;
       }
+      if (initScores[0] >= getParams().similarityCutoff) {
+        initHit.reset(new ROMol(hit));
+      }
     } else {
       initScores[0] = -1.0;  // To flag it as a bad hit.
     }
@@ -1208,9 +1215,11 @@ bool SynthonSpaceShapeSearcher::verifyHit(
       }
     }
   }
-  if (!foundHit && initScores[0] > getParams().similarityCutoff) {
-    // Stick with what we found at the start, which should still be in hit,
-    // because the conformational expansion didn't do any better.
+  if (!foundHit && initHit && initScores[0] >= getParams().similarityCutoff) {
+    // Stick with what we found at the start because the conformational
+    // expansion didn't do any better.
+    hit = std::move(*initHit);
+    updateBestHitSoFar(hit, initScores[0]);
     foundHit = true;
   }
   return foundHit;
diff --git a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSubstructureSearcher.cpp b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSubstructureSearcher.cpp
index e25057544..af48b6a79 100644
--- a/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSubstructureSearcher.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/SynthonSpaceSubstructureSearcher.cpp
@@ -392,10 +392,11 @@ bool SynthonSpaceSubstructureSearcher::extraSearchSetup(
     // For the fingerprints, ring info is required, and this needs to be
     // done for every copy of the fragment. We also need to fix any
     // query atoms.
-    for (const auto &frag : frags) {
-      unsigned int otf;
-      sanitizeMol(*static_cast<RWMol *>(frag), otf, MolOps::SANITIZE_SYMMRINGS);
-
+    for (auto &frag : frags) {
+      if (!frag->getRingInfo()->isInitialized()) {
+        VECT_INT_VECT arings;
+        MolOps::symmetrizeSSSR(*frag, arings);
+      }
       // Query atoms may define the environment of the fragment (via recursive
       // SMARTS, for example) that a potentially matching synthon may not
       // have, so they need to be made more generic.  For example, if the
@@ -407,7 +408,7 @@ bool SynthonSpaceSubstructureSearcher::extraSearchSetup(
       // combined with a synthon such as [*1]ccccc[*3].  The downside is that
       // the fragment will become less discriminating leading to more false
       // positives in the initial screenout, hence the warning.
-      if (details::removeQueryAtoms(*static_cast<RWMol *>(frag)) &&
+      if (details::removeQueryAtoms(*reinterpret_cast<RWMol *>(frag)) &&
           !saidSomething) {
         saidSomething = true;
         BOOST_LOG(rdInfoLog) << "Complex queries can be slow." << std::endl;
diff --git a/Code/GraphMol/SynthonSpaceSearch/Wrap/testSynthonSpaceSearch.py b/Code/GraphMol/SynthonSpaceSearch/Wrap/testSynthonSpaceSearch.py
index 61b0dc22e..78d443413 100644
--- a/Code/GraphMol/SynthonSpaceSearch/Wrap/testSynthonSpaceSearch.py
+++ b/Code/GraphMol/SynthonSpaceSearch/Wrap/testSynthonSpaceSearch.py
@@ -413,7 +413,6 @@ class TestCase(unittest.TestCase):
       restartFile = tmp.name
     os.unlink(restartFile)
     textFile = self.sssDir / "amide_space.txt"
-    print(f"Restart : {restartFile}  from {textFile}")
 
     p = multiprocessing.Process(target=createShapeDatabaseForTest, args=(textFile, restartFile))
     p.start()
diff --git a/Code/GraphMol/SynthonSpaceSearch/fingerprint_search_catch_tests.cpp b/Code/GraphMol/SynthonSpaceSearch/fingerprint_search_catch_tests.cpp
index e58ae026c..a45e78b93 100644
--- a/Code/GraphMol/SynthonSpaceSearch/fingerprint_search_catch_tests.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/fingerprint_search_catch_tests.cpp
@@ -145,8 +145,6 @@ TEST_CASE("FP Binary File") {
   SearchResults results;
   auto queryMol = "O=C(Nc1c(CNC=O)cc[s]1)c1nccnc1"_smiles;
   SynthonSpaceSearchParams params;
-  std::cout << params.similarityCutoff << " and "
-            << params.approxSimilarityAdjuster << std::endl;
   for (auto numThreads : std::vector<int>{1, 2, -1}) {
     synthonspace.readDBFile(libName, numThreads);
     params.numThreads = numThreads;
diff --git a/Code/GraphMol/SynthonSpaceSearch/shape_search_catch_tests.cpp b/Code/GraphMol/SynthonSpaceSearch/shape_search_catch_tests.cpp
index bdc3bbc6e..231a7321a 100644
--- a/Code/GraphMol/SynthonSpaceSearch/shape_search_catch_tests.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/shape_search_catch_tests.cpp
@@ -101,7 +101,6 @@ TEST_CASE("Shape Small test") {
   params.bestHit = true;
   auto queryMol = v2::SmilesParse::MolFromSmiles(querySmi);
   auto results = synthonspace.shapeSearch(*queryMol, params);
-  unsigned int j = 0;
   CHECK(results.getHitMolecules().size() == 3);
   for (const auto &mol : results.getHitMolecules()) {
     // Different machine/compiler combinations give slightly different results
@@ -113,7 +112,6 @@ TEST_CASE("Shape Small test") {
     auto scores = GaussianShape::ScoreMolecule(*queryMol, *mol);
     CHECK_THAT(mol->getProp<double>("Similarity"),
                Catch::Matchers::WithinAbs(scores[0], 0.001));
-    ++j;
   }
 }
 
@@ -165,6 +163,7 @@ TEST_CASE("Shape DB Writer") {
   synthonspace.readTextFile(libName, cancelled);
   CHECK(synthonspace.getNumReactions() == 1);
   ShapeBuildParams shapeBuildParams;
+  shapeBuildParams.timeOut = 0;
   synthonspace.buildSynthonShapes(cancelled, shapeBuildParams);
 
   auto spaceName = std::tmpnam(nullptr);
@@ -180,6 +179,7 @@ TEST_CASE("Shape DB Writer") {
   for (size_t i = 0; i < irxn->getSynthons().size(); ++i) {
     REQUIRE(irxn->getSynthons()[i].size() == orxn->getSynthons()[i].size());
     for (size_t j = 0; j < irxn->getSynthons()[i].size(); ++j) {
+      REQUIRE(irxn->getSynthons()[i][j].second->getShapes());
       REQUIRE(irxn->getSynthons()[i][j]
                   .second->getShapes()
                   ->getShapes()
diff --git a/Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp b/Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp
index fe1e597f1..1a71dc83d 100644
--- a/Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp
+++ b/Code/GraphMol/SynthonSpaceSearch/substructure_search_catch_tests.cpp
@@ -512,6 +512,7 @@ TEST_CASE("Greg Space Failure") {
 
   SubstructMatchParameters matchParams;
   SynthonSpaceSearchParams params;
+  params.timeOut = 0;
   auto results =
       synthonspace.substructureSearch(*queryMol, matchParams, params);
   CHECK(results.getHitMolecules().size() == 1);
