diff --git a/External/ChemDraw/CMakeLists.txt b/External/ChemDraw/CMakeLists.txt
index b75d32795..738db8f46 100644
--- a/External/ChemDraw/CMakeLists.txt
+++ b/External/ChemDraw/CMakeLists.txt
@@ -55,6 +55,15 @@ if(RDK_BUILD_CHEMDRAW_SUPPORT)
   # however, we don't want to install it so use the undocumented EXCLUDE_FROM_ALL
   add_subdirectory(chemdraw/expatpp EXCLUDE_FROM_ALL)
 
+  # our windows build has trouble linking to expatpp because of the expatpp.h
+  # include. Make the path to it a public dependency so that things that
+  # depend on it can find it.
+  target_include_directories(expatpp PUBLIC ${EXPAT_INCLUDE_DIR})
+
+  # Linking against static expat requires XML_STATIC to be defined
+  target_compile_definitions(expatpp PUBLIC XML_STATIC)
+
+
   include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}/chemdraw/expatpp/expatpp-code-r6-trunk/src_pp
     ${CMAKE_CURRENT_SOURCE_DIR}/chemdraw/expatpp/expatpp-code-r6-trunk/expat/lib)
